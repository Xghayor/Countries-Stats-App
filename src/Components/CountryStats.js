import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import { countryData } from '../Redux/Countries/countriesSlice';
import './styles/CountryStats.css';

const CountryStats = () => {
  const { searchTerm } = useParams();
  const { countries, loading, error } = useSelector((state) => state.countries);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(countryData(searchTerm));
  }, [dispatch, searchTerm]);

  if (loading) {
    return <div className="loader" />;
  }

  if (error) {
    return (
      <div>
        Error: Not Found
        {error.message}
      </div>
    );
  }

  if (!countries) {
    return null;
  }

  return (
    <div className="country-statistics">
      <h5 className="stats-heading">
        Country Statistics for
        {' '}
        {searchTerm}
      </h5>
      {countries.map((country) => (
        <div key={country.name} className="country-card">
          <div className="info-row dark-color">
            <p className="label">GDP:</p>
            <p className="value">{country.gdp}</p>
          </div>
          <div className="info-row  light-color">
            <p className="label">Sex Ratio:</p>
            <p className="value">{country.sex_ratio}</p>
          </div>
          <div className="info-row dark-color">
            <p className="label">Surface Area:</p>
            <p className="value">{country.surface_area}</p>
          </div>
          <div className="info-row  light-color">
            <p className="label">Life Expectancy (Male):</p>
            <p className="value">{country.life_expectancy_male}</p>
          </div>
          <div className="info-row dark-color">
            <p className="label">Unemployment Rate:</p>
            <p className="value">{country.unemployment}</p>
          </div>
          <div className="info-row  light-color">
            <p className="label">Imports:</p>
            <p className="value">{country.imports}</p>
          </div>
          <div className="info-row dark-color">
            <p className="label">Homicide Rate:</p>
            <p className="value">{country.homicide_rate}</p>
          </div>
          <div className="info-row  light-color">
            <p className="label">Tourists:</p>
            <p className="value">{country.tourists}</p>
          </div>
          <div className="info-row dark-color">
            <p className="label">Exports:</p>
            <p className="value">{country.exports}</p>
          </div>
          <div className="info-row light-color">
            <p className="label">Capital:</p>
            <p className="value">{country.capital}</p>
          </div>
          <div className="info-row dark-color">
            <p className="label">Currency Code:</p>
            <p className="value">{country.currency.code}</p>
          </div>
          <div className="info-row light-color">
            <p className="label">Currency Name:</p>
            <p className="value">{country.currency.name}</p>
          </div>
          <div className="info-row dark-color">
            <p className="label">Employment in Industry:</p>
            <p className="value">{country.employment_industry}</p>
          </div>
          <div className="info-row light-color">
            <p className="label">Population:</p>
            <p className="value">{country.population}</p>
          </div>
        </div>
      ))}
    </div>
  );
};

export default CountryStats;
