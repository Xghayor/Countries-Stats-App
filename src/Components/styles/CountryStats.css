.country-statistics {
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
  font-size: 16px;
  color: #333;
  background-color: rgba(46, 128, 228, 0.772);
}

.country-card {
  border: 1px solid grey;
  margin-bottom: 20px;
}

.info-row {
  display: flex;
  justify-content: space-between;
  margin-bottom: 8px;
  align-items: center;
}

.label {
  font-weight: bold;
  color: white;
  margin: 0;
}

.value {
  color: white;
  margin-right: 40px;
  text-align: left;
}

.stats-heading {
  font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;
  font-size: 24px;
  margin: 0;
  color: white;
  background-color: rgb(0, 68, 170);
  padding: 10px 20px;
  z-index: 999;
  box-shadow: 0 0 10px rgba(0, 0, 255, 0.2);
}

.loader {
  border: 6px solid #f3f3f3;
  border-top: 6px solid #3498db;
  border-radius: 50%;
  width: 40px;
  height: 40px;
  margin: 5% auto;
  animation: spin 2s linear infinite;
}

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}

.light-color {
  font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
  font-size: 18px;
  background-color: rgba(37, 56, 131, 0.527);
  margin: 0;
  height: 60px;
  color: white;
  padding: 20px 0 0 20px;
}

.dark-color {
  font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
  font-size: 18px;
  background-color: rgba(19, 19, 165, 0.683);
  margin: 0;
  height: 60px;
  color: white;
  padding: 20px 0 0 20px;
}
