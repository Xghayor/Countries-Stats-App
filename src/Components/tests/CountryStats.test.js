import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import CountryStats from '../Countries';

const initialState = {
  countries: {
    countries: [],
    loading: false,
    error: null,
  },
};

const mockStore = configureStore([]);
const store = mockStore(initialState);

test('CountryStats component renders correctly', () => {
  const { container } = render(
    <Provider store={store}>
      <MemoryRouter>
        <CountryStats />
      </MemoryRouter>
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
