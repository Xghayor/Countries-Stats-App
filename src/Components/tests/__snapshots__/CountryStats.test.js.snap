// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryStats component renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <h4
      class="grid-title"
    >
      STATS BY COUNTRY
    </h4>
    <ul
      class="grid"
    >
      <li
        class="grid-item dark-blue"
      >
        <div
          class="grid-item-box"
        >
          <img
            alt="Shared Flag"
            class="country-image"
            src="France.png"
          />
          <a
            class="view-button"
            href="/country/United States"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
              />
            </svg>
          </a>
          <div
            class="country-name"
          >
            United States
          </div>
        </div>
      </li>
      <li
        class="grid-item light-blue"
      >
        <div
          class="grid-item-box"
        >
          <img
            alt="Shared Flag"
            class="country-image"
            src="France.png"
          />
          <a
            class="view-button"
            href="/country/France"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
              />
            </svg>
          </a>
          <div
            class="country-name"
          >
            France
          </div>
        </div>
      </li>
      <li
        class="grid-item dark-blue"
      >
        <div
          class="grid-item-box"
        >
          <img
            alt="Shared Flag"
            class="country-image"
            src="France.png"
          />
          <a
            class="view-button"
            href="/country/Canada"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
              />
            </svg>
          </a>
          <div
            class="country-name"
          >
            Canada
          </div>
        </div>
      </li>
      <li
        class="grid-item light-blue"
      >
        <div
          class="grid-item-box"
        >
          <img
            alt="Shared Flag"
            class="country-image"
            src="France.png"
          />
          <a
            class="view-button"
            href="/country/Germany"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
              />
            </svg>
          </a>
          <div
            class="country-name"
          >
            Germany
          </div>
        </div>
      </li>
      <li
        class="grid-item dark-blue"
      >
        <div
          class="grid-item-box"
        >
          <img
            alt="Shared Flag"
            class="country-image"
            src="France.png"
          />
          <a
            class="view-button"
            href="/country/China"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
              />
            </svg>
          </a>
          <div
            class="country-name"
          >
            China
          </div>
        </div>
      </li>
      <li
        class="grid-item light-blue"
      >
        <div
          class="grid-item-box"
        >
          <img
            alt="Shared Flag"
            class="country-image"
            src="France.png"
          />
          <a
            class="view-button"
            href="/country/Qatar"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
              />
            </svg>
          </a>
          <div
            class="country-name"
          >
            Qatar
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
